src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -380,12 +415,14 @@ public void StopProcessingNextRequest(bool serverInitiated)
             Log.HPackDecodingError(ConnectionId, _currentHeadersStream.StreamId, ex);
             error = ex;
             errorCode = Http2ErrorCode.COMPRESSION_ERROR;
+            reason = ConnectionEndReason.ErrorReadingHeaders;
         }
         catch (Exception ex)
         {
             Log.LogWarning(0, ex, CoreStrings.RequestProcessingEndError);
             error = ex;
             errorCode = Http2ErrorCode.INTERNAL_ERROR;
+            reason = ConnectionEndReason.UnexpectedError;
```

Comment:
Maybe "unknown error"?  It seems weird to describe it as "unexpected".

Reply:
I don't like the word unknown because we do know it, it just doesn't fall into one of the known reasons we have for a connection ending.

What about `OtherError`? Or just `Other`?

Reply:
Seems error-like, so probably `OtherError`.  If it's not always error-like, I'm fine with `Other`.